src/man/sssd-ldap.5.xml

```diff
@@ -161,6 +161,45 @@
                     </listitem>
                 </varlistentry>
 
+                <varlistentry>
+                    <term>ldap_read_rootdse (string)</term>
+                    <listitem>
+                        <para>
+                            SSSD reads RootDSE to get information about LDAP and
+                            its capabilities. By default, this is done
+                            anonymously. However, this may not be permitted by
+                            the LDAP server. In such cases we can use this option
+                            to influence SSSD behavior.
+                        </para>
+                        <para>
+                            Allowed values are:
+                            <itemizedlist>
+                                <listitem>
+                                    <para>
+                                        anonymous
+                                    </para>
+                                </listitem>
+                                <listitem>
+                                    <para>
+                                        authenticated
+                                    </para>
+                                </listitem>
+                                <listitem>
+                                    <para>
+                                        never
+                                    </para>
+                                </listitem>
+                            </itemizedlist>
+                            Note that when using the "anonymous" option, SSSD
+                            may attempt to read RootDSE after authentication
+                            if anonymous access fails.
```

Comment:
I don't quite understand this statement. Can you elaborate?

Reply:
Is it related to the comment in the code below?

         /* We did not read rootDSE during unauthenticated bind becase
         * it is unaccessible for anonymous user or because
         * ldap_read_rootdse is set to "authenticated"
         * Let's try to read it now */

Reply:
Yes it is related. The "anonymous" option basically enables current/original SSSD behavior.  In this mode SSSD tries to read RootDSE anonymously but if that fails SSSD attempts to read it after authentication once again.

Failures are not critical - we can proceed without reading RootDSE if everything (like e. g. authentication mechanism) is set correctly manually in sssd.conf.

This option allows us to skip the attempts that would fail anyway due to LDAP server restrictions/custom setup and get rid of annoying error messages in the log in such environment.

Reply:
Thanks for clarification. IMO the wording can be improved a bit. Something like

`By default, using the "anonymous" option, SSSD tries to read RootDSE anonymously. If this fails SSSD retries the attempt with authentication.`

Reply:
Thanks, updated.